internal/model/user_server.go

```diff
@@ -165,6 +165,28 @@ func (this *ServerUser) ToCourseUser(courseID string) (*CourseUser, error) {
 	return courseUser, courseUser.Validate()
 }
 
+// Convert this server user into a course super user for the specific course.
+// The LMSID field will be nil if the user is not enrolled in the given course.
+func (this *ServerUser) ToCourseSuperUser(courseID string) (*CourseUser, error) {
+	courseUser, err := this.ToCourseUser(courseID)
+	if err != nil {
+		return courseUser, err
+	}
+
+	if courseUser == nil {
+		courseUser = &CourseUser{
+			Email: this.Email,
+			Name:  this.Name,
+			Role:  CourseRoleSuper,
```

Comment:
You could probably remove this since the role is being added on a few lines down anyway.